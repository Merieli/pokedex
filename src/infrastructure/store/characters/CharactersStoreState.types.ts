export interface CharactersStoreState {
    characters: any[];
}
