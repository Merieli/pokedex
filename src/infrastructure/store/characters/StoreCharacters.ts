import { defineStore } from 'pinia';
import { CharactersStoreState } from './CharactersStoreState.types';
import getAllCharacters from './actions/getAllCharacters';

const state = (): CharactersStoreState => ({
    characters: [],
});

const getters = {};

const actions = {
    getAllCharacters,
};

export const useCounterStore = defineStore('characters', {
    state,
    actions,
});
