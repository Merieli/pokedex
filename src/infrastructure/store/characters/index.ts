export * from './CharactersStoreState.types';
export * from './StoreCharacters';
