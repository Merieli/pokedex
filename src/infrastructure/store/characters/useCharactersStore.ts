import { Ref, ref } from 'vue';

import { Character } from '@/domain';
import CharactersGatewayHttp from '@/infrastructure/gateway/CharactersGatewayHttp';

import { defineStore } from 'pinia';

export const useCharactersStore = defineStore('characters', () => {
    const characters: Ref<Character[]> = ref([]);
    const isLoading: Ref<boolean> = ref(false);

    const getAllCharacters = async (): Promise<void> => {
        try {
            isLoading.value = true;
            const api = new CharactersGatewayHttp();
            const response = await api.getAll();
            characters.value = response.results;
        } catch (error) {
            // const { code } = logError('Action getAllCharacters', error);
            // notifyUser(code, 'imprimir todos os personagens')  //função para exibir o erro para o usuário
        } finally {
            isLoading.value = false;
        }
    };

    return {
        characters,
        isLoading,
        getAllCharacters,
    };
});
