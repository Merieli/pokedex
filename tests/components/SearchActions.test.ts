import SearchActionsVue from '@components/SearchActions.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { createVuetify } from 'vuetify';
import { mockStoreCharacters } from '../__mocks__/mockStoreCharacters';

import { useCharactersStore } from '@/infrastructure/store/characters';

describe('SearchActions.vue', () => {
    const setupWrapper = () => {
        const pinia = createTestingPinia({
            initialState: {
                characters: {
                    pagination: {
                        currentPage: 1,
                        total: 1,
                        results: 10,
                    },
                    charactersPerPage: mockStoreCharacters,
                },
            },
        });
        const vuetify = createVuetify();

        return {
            wrapper: mount(SearchActionsVue, {
                global: {
                    plugins: [pinia, vuetify],
                },
            }),
            store: useCharactersStore(),
        };
    };

    describe('Integração/Componente', () => {
        describe('👀 Renderização:', () => {
            test('Dado o html do componente quando renderizado então deve ter os mesmos dados do snapshot gravado', () => {
                const { wrapper } = setupWrapper();

                expect(wrapper.html()).toMatchSnapshot();
            });

            test('Dado o componente Quando renderizado Então deve possuir um componente Vuetify de texto para o campo de busca', async () => {
                const { wrapper } = setupWrapper();
                const searchInput = wrapper.find('v-text-field');

                expect(searchInput.exists()).toBeTruthy();
            });

            test('Dado o componente Quando renderizado Então deve possuir um componente Vuetify de botão para gerar personagem aleatóriamente', async () => {
                const { wrapper } = setupWrapper();
                const searchButton = wrapper.find('v-btn');

                expect(searchButton.exists()).toBeTruthy();
            });

            test('Dado o componente Quando renderizado Então deve possuir um componente Vuetify de icone no botão para gerar personagem aleatóriamente', async () => {
                const { wrapper } = setupWrapper();
                const searchButtonIcon = wrapper.find('v-icon');

                expect(searchButtonIcon.exists()).toBeTruthy();
            });
        });

        describe('🧠 Comportamento:', () => {
            test('Dado um botão para gerar um personagem aleatório Quando clicado Então deve chamar a action necessária', async () => {
                const { wrapper, store } = setupWrapper();
                const spyRandomCharacter = vi.spyOn(store, 'generateRandomCharacter');
                const buttonRandom = wrapper.find('[data-search-actions="button"]');

                await buttonRandom.trigger('click');

                expect(spyRandomCharacter).toHaveBeenCalledTimes(1);
            });
        });
    });
});
