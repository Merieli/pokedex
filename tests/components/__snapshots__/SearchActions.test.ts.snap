// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchActions.vue > Integra√ß√£o/Componente > üëÄ Renderiza√ß√£o: > Dado o html do componente quando renderizado ent√£o deve ter os mesmos dados do snapshot gravado 1`] = `
"<div class=\\"search-actions\\" data-v-670007cd=\\"\\">
  <v-text-field class=\\"search-actions__input\\" data-search-actions=\\"input\\" append-inner-icon=\\"mdi-magnify\\" label=\\"Pesquise por um personagem\\" placeholder=\\"Ex: Morty Smith\\" density=\\"compact\\" variant=\\"solo\\" single-line=\\"\\" hide-details=\\"\\" clearable=\\"\\" data-v-670007cd=\\"\\"></v-text-field>
  <v-btn class=\\"search-actions__button\\" elevation=\\"1\\" depressed=\\"\\" data-v-670007cd=\\"\\"> Generate random <v-icon class=\\"search-actions__icon-button\\" icon=\\"mdi-reload\\" right=\\"\\" data-v-670007cd=\\"\\"></v-icon>
  </v-btn>
</div>"
`;
